velox/type/Timestamp.cpp

```diff
@@ -278,16 +282,40 @@ std::string Timestamp::tmToString(
   appendSmallInt(tmValue.tm_min, out);
   out += ':';
   appendSmallInt(tmValue.tm_sec, out);
-  out += '.';
-  int offset = out.size();
   if (options.precision == TimestampToStringOptions::Precision::kMilliseconds) {
     nanos /= 1'000'000;
+  } else if (
+      options.precision == TimestampToStringOptions::Precision::kMicroseconds) {
+    nanos /= 1'000;
+  }
+  if (options.skipTrailingZeros && nanos == 0) {
+    return out;
   }
-  while (nanos > 0) {
-    out += '0' + nanos % 10;
-    nanos /= 10;
+  out += '.';
+  const int offset = out.size();
+  int trailingZeros = 0;
+
+  if (options.skipTrailingZeros) {
+    while (nanos > 0) {
+      if (out.size() == offset && nanos % 10 == 0) {
+        trailingZeros += 1;
+      } else {
+        out += '0' + nanos % 10;
+      }
+      nanos /= 10;
+    }
+  } else {
+    while (nanos > 0) {
+      if (nanos % 10 == 0) {
```

Comment:
There is no need to check if it is zero here, just keep the original one liner

Reply:
Nice catch. Opened https://github.com/facebookincubator/velox/pull/8250 to address this comment, thanks.